src/widget/systemtrayicon.cpp

```diff
@@ -233,13 +244,17 @@ void SystemTrayIcon::setContextMenu(QMenu* menu)
                 };
                 QImage* image = new QImage(a->icon().pixmap(64, 64).toImage());
                 if (image->format() != QImage::Format_RGBA8888_Premultiplied)
-                *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
+                    *image = image->convertToFormat(QImage::Format_RGBA8888_Premultiplied);
                 GdkPixbuf* pixbuf = gdk_pixbuf_new_from_data(image->bits(), GDK_COLORSPACE_RGB, image->hasAlphaChannel(),
```

Comment:
Phew, this is is total brain f**k. You're right, the callback reinterpret_casts gpointer to QImage* and free's it. But… we don't actually need the callback here don't you think? As this is GTK+ code, we should rather use completely their classes and replace the `QImage` code here IMO.